src/codeu/chat/server/Model.java

```diff
@@ -68,63 +72,271 @@ public int compare(Time a, Time b) {
+    if(where != null)
+      query += " where " + where;
+    if(orderBy != null)
+      query += " ORDER BY UNAME " + orderBy;
+    query += ";";
+    return dbConnection.dbQueryUsers(query);
   }
 
   public Uuid userGeneration() {
     return currentUserGeneration;
   }
 
   public void add(Conversation conversation) {
-    conversationById.insert(conversation.id, conversation);
-    conversationByTime.insert(conversation.creation, conversation);
-    conversationByText.insert(conversation.title, conversation);
+
+    try {
+      dbConnection.dbUpdate("INSERT INTO CONVERSATIONS (ID,CNAME,OWNERID,TimeCreated) " +
+              "VALUES (" + SQLFormatter.sqlID(conversation.id) + ", " + SQLFormatter.sqlName(conversation.title) + ", " +
+              SQLFormatter.sqlID(conversation.owner) + ", " + SQLFormatter.sqlCreationTime(conversation.creation) + ");");
+
+      LOG.info("Conversation added: " + conversation.id);
+    } catch (Exception e) {
+      LOG.info(
+              "newConversation fail - Verify connection and try again shortly");
+      System.err.println(e.getClass().getName() + ": " + e.getMessage());
+      System.exit(0);
+    }
+
+    add(conversation.owner, conversation.id);
+  }
+
+  // Add user to a conversation
+  public void add(Uuid user, Uuid conversation) {
+    try {
+      dbConnection.dbUpdate("INSERT INTO USER_CONVERSATION (ID, USERID, CONVERSATIONID) " +
+                            "VALUES (" + SQLFormatter.sqlID(user, conversation) + ", " + SQLFormatter.sqlID(user) + ", " + SQLFormatter.sqlID(conversation) + ");");
+    } catch (Exception e) {
+      LOG.info(
+              "Adding user to conversation fail - Verify connection and try again shortly");
+      System.err.println(e.getClass().getName() + ": " + e.getMessage());
+      System.exit(0);
+    }
+  }
+
+  public void update(Conversation conversation) {
+    try {
+      dbConnection.dbUpdate("UPDATE CONVERSATIONS set" +
+              " CNAME = " + SQLFormatter.sqlName(conversation.title) + ", " +
+              " OWNERID = " + SQLFormatter.sqlID(conversation.owner) + ", " +
+              " TimeCreated = " + SQLFormatter.sqlCreationTime(conversation.creation) +
+              " where ID = " + SQLFormatter.sqlID(conversation.id) +
+              ";");
+      LOG.info(
+              "updateConversation success (conversation.id=%s conversation.name=%s conversation.time=%s)",
+              conversation.id,
+              conversation.title,
+              conversation.creation);
+    } catch (Exception e) {
+      LOG.info(
+              "updateConversation fail - Database update error (conversation.id=%s conversation.name=%s conversation.time=%s)",
+              conversation.id,
+              conversation.title,
+              conversation.creation);
+      System.err.println(e.getClass().getName() + ": " + e.getMessage());
+      System.exit(0);
+    }
+  }
+
+  public Collection<Conversation> conversationById(String where, String orderBy) {
```

Comment:
Having these methods take strings leaks details about the model. It is ok if you use these as private helper methods, but the public methods should be using objects like Uuid and you should use an enum to signal ordering if you really need to expose that.